clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -653,11 +654,23 @@ void PointerVariableConstraint::setTypedef(TypedefNameDecl* T, std::string s) {
 // variables and potentially nested function pointer declaration. Produces a
 // string that can be replaced in the source code.
 
+std::string PointerVariableConstraint::gatherQualStrings(void) const {
+  std::ostringstream S;
+  uint32_t Idx = 0;
+
+  for (auto It = Vars.begin(); It != Vars.end(); It++, Idx++) {
+    getQualString(Idx, S);
+  }
+
+  return S.str();
```

Comment:
It looks like this could generate i.e. "const const" if you have `const int * const * a` or something like that. So the function name is appropriate. But it seems to be used below as a single qualifier. Is that correct?